spec/lib/tasks/after_tasks.rake_spec.rb

```diff
@@ -503,3 +503,51 @@
     end
   end
 end
+
+describe "rake After:migrate_pinch_request_signup" do
+  context "for an assignment with a request_signup_id" do
+    let(:assignment) { create(:challenge_assignment) }
+
+    it "does nothing" do
+      expect do
+        subject.invoke
+      end.to avoid_changing { assignment.reload.request_signup_id }
+        .and output("Migrated pinch_request_signup for 0 challenge assignments.\n").to_stdout
+    end
+  end
+
+  context "for an assignment with a request_signup_id and a pinch_request_signup_id" do
+    let(:collection) { create(:collection) }
+    let(:assignment) do
+      create(:challenge_assignment,
+             collection: collection,
+             pinch_request_signup_id: create(:challenge_signup, collection: collection).id)
+    end
+
+    it "does nothing" do
+      expect do
+        subject.invoke
+      end.to avoid_changing { assignment.reload.request_signup_id }
+        .and output("Migrated pinch_request_signup for 0 challenge assignments.\n").to_stdout
+    end
+  end
+
+  context "for an assignment without a request_signup_id and a pinch_request_signup_id" do
```

Comment:
Minor thing, to me this sounds like "an assignment with neither a request_signup_id nor a pinch_request_signup_id". Maybe something like the following?
```suggestion
  context "for an assignment with a pinch_request_signup_id but no request_signup_id" do
```